homeassistant/components/sentry/config_flow.py

```diff
@@ -0,0 +1,58 @@
+"""Config flow for sentry integration."""
+import logging
+
+from sentry_sdk.utils import BadDsn, Dsn
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+
+from .const import CONF_DSN, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_DSN): str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the DSN input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    # validate the dsn
+    Dsn(data["dsn"])
```

Comment:
Is this doing I/O? If so, run it inside the executor `await hass.async_add_executor_job(Dsn, data["dsn"])`

Reply:
No I/O here - just validation on the input value (does it look like XYZ)